BlazorApp/Program.cs

```diff
@@ -12,10 +15,17 @@ public static async Task Main(string[] args)
             builder.RootComponents.Add<App>("#app");
             builder.RootComponents.Add<HeadOutlet>("head::after");
 
+            builder.Services.AddBlazoredLocalStorage();
+            builder.Services.AddAuthorizationCore();
+
             builder.Services.AddScoped(sp => new HttpClient { BaseAddress = new Uri("https://localhost:7036/") });//api https address
 
             //Add services IUserService,... from Shared-Layer
             builder.Services.AddScoped<IUserServices, UserServices>();
+            builder.Services.AddScoped<IAuthenticationService, AuthenticationService>();
+            builder.Services.AddScoped<AuthenticationStateProvider, CustomAuthStateProvider>();
+            builder.Services.AddScoped<CustomAuthStateProvider>();
```

Comment:
Yours aprouch create two separate instances of CustomAuthStateProvider, leading to potential inconsistencies.
Change to:
builder.Services.AddScoped<CustomAuthStateProvider>();
builder.Services.AddScoped<AuthenticationStateProvider>(provider =>
provider.GetRequiredService<CustomAuthStateProvider>());

This approach  guarantees that AuthenticationStateProvider and CustomAuthStateProvider always reference the same instance, ensuring that any updates to authentication state are immediately visible throughout the application.